jenkins/Hono-Release-Pipeline.groovy

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env groovy
+
+/*******************************************************************************
+ * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+/**
+ * Jenkins pipeline script for Hono release.
+ *
+ */
+
+node {
+    def utils = evaluate readTrusted("jenkins/Hono-PipelineUtils.groovy")
+    properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '3')), parameters([
+            string(defaultValue: '',
+                    description: "The branch to build and release from.\nExamples: \n master\n1.0.x\n1.x",
+                    name: 'BRANCH',
+                    trim: true),
+            string(defaultValue: '',
+                    description: "The version identifier to use for the artifacts built and released by this job. \nExamples:\n1.0.0-M6\n1.0.0-RC1\n2.1.0",
+                    name: 'RELEASE_VERSION',
+                    trim: true),
+            string(defaultValue: '',
+                    description: "The version identifier to use during development of the next version.\nExamples:\n2.0.0-SNAPSHOT\n1.1.0-SNAPSHOT",
+                    name: 'NEXT_VERSION',
+                    trim: true),
+            string(defaultValue: '',
+                    description: "The version identifier to use during development of the next version.\nExamples:\n2.0.0-SNAPSHOT\n1.1.0-SNAPSHOT",
```

Comment:
I do not think that this is the correct description of the parameter, is it?
How will the user who starts the pipeline know the value to set for this parameter? When creating a build job manually, you can select the credentials from a drop down box. However, since this seems to be a build parameter, the user will probably need to enter the value manually ...

Reply:
I think you are referring to the credentials parameter. I will use that instead of the current one.

Reply:
???

Reply:
Ok, so you change the *type* of the parameter from `String` to `Credentials`, right?

Reply:
yes, along with suitable arguments